### PR TITLE
[ios] Update changelog for v4.1.0-beta.1

### PR DESCRIPTION
**Pull request by fabian-guerra**
Podspec bump and changelog updates.

### CODE REVIEW

**Comment by friedbunny** [comment]
There are a variety of core changes that weren’t (and should have been) changelog’d — see the [`needs changelog`](https://github.com/mapbox/mapbox-gl-native/pulls?q=is%3Apr+is%3Aclosed+label%3A%22needs+changelog%22) tag for some of the ones I’ve noticed recently.

**Comment by fabian-guerra** [comment]
For some reason the podspec bump got caught in CP this https://github.com/mapbox/mapbox-gl-native/pull/12087

File: platform/ios/CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +23,9 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
 * Added custom `-hitTest:withEvent:` to `MGLSMCalloutView` to avoid registering taps in transparent areas of the standard annotation callout. ([#11939](https://github.com/mapbox/mapbox-gl-native/pull/11939))
 * Improved performance and memory impact of `MGLScaleBar`. ([#11921](https://github.com/mapbox/mapbox-gl-native/pull/11921))
 * Fixed race conditions that could cause crashes when re-using `MGLMapSnapshotter` or using multiple snapshotters at the same time. ([#11831](https://github.com/mapbox/mapbox-gl-native/pull/11831))
+* Fixed an issue where an empty `MGLFeature`s array caused high CPU utilization. ([#11985](https://github.com/mapbox/mapbox-gl-native/pull/11985))
+* Improved offline download. ([#11284](https://github.com/mapbox/mapbox-gl-native/pull/11284))
```
Comment (friedbunny) [inline platform/ios/CHANGELOG.md]: Let’s be a little more specific and say: “download _performance_.”

File: platform/ios/CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +23,9 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
...
+* Fixed an issue that caused `MGLMapView visibleFeaturesAtPoint:]` return an empty features array when removing/adding features. ([#12076](https://github.com/mapbox/mapbox-gl-native/pull/12076))
```
Comment (friedbunny) [inline platform/ios/CHANGELOG.md]: A few small tweaks:

> Fixed an issue that caused `-[MGLMapView visibleFeaturesAtPoint:]` to return an empty features array when adding or removing features.

File: platform/ios/CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
 ### Style layers
 
 * Added support for aggregate expressions as input values to `MGL_MATCH` expressions. ([#11866](https://github.com/mapbox/mapbox-gl-native/pull/11866))
+* Fixed a crash caused by `literal` expressions when used in non-data-driven properties. ([#11960](https://github.com/mapbox/mapbox-gl-native/pull/11960))
```
Comment (friedbunny) [inline platform/ios/CHANGELOG.md]: This fix made it into `ios-v4.0.2` (via #11961), so we don’t need to changelog it here.

**Comment by friedbunny** [comment]
Found another one: this line should link to its GitHub issue, so we need to add a link to #11784.

https://github.com/mapbox/mapbox-gl-native/blame/c48bdcde7553b34dc9dbbd0c41cb31997044b0bd/platform/ios/CHANGELOG.md#L20

**Comment by friedbunny** [comment]
https://github.com/mapbox/mapbox-gl-native/pull/12100 (https://github.com/mapbox/mapbox-gl-native/pull/12076) also needs a changelog.

**Comment by 1ec5** [comment]
Going forward, changes such as these need to be merged to master first, then cherry-picked to the release branch.

**Pull request merged**